File: Backend/PrestationServices/src/main/java/tn/esprit/Entities/Entreprise.java
package tn.esprit.Entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.util.List;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Entreprise {
    @Id
    @GeneratedValue(strategy= GenerationType.IDENTITY)
    private int identreprise ;
    private String nom ;
    private String secteur;
    private String adresse ;
    private int telephone;
    private String latitude;
    private String longitude;
    private String description;
    private int invitations;

    @OneToMany(mappedBy = "entreprise1")
    @JsonIgnore
    private List<Partenaire> partenaires;
    @OneToMany(mappedBy = "entreprise2")
    @JsonIgnore
    private List<Partenaire> partenairesB;

    @OneToMany(mappedBy = "entreprise")
    @JsonIgnore
    List<BesoinCommercial> besoinCommercials ;

    @OneToMany(mappedBy = "entreprise")
    @JsonIgnore
    private List<ServiceCommercial> serviceCommercials;
    @OneToMany(mappedBy = "entrepriseB")
    @JsonIgnore
    private List<Connexion> connexionsEntrantes;



}

File: Backend/Frontend/src/app/models/Entreprise/entreprise.ts
import { BesoinCommercial } from "../BesoinCommercial/besoin-commercial";
import { Connexion } from "../Connexion/connexion";
import { Partenaire } from "../Partenaire/partenaire";
import { ServiceCommercial } from "../ServiceCommercial/service-commercial";
import { BesoinOffre } from "../besoin-offre";
import { Reunion } from "../reunion.model";

export class Entreprise {
    identreprise!: number;
    nom!: string;
    secteur!: string;
    adresse!: string;
    telephone!: number;
    latitude!:string;
    longitude!:string;
    description!:string;
    invitations!: number;
    partenaires!: Partenaire[];
    partenairesB!: Partenaire[];
    besoinCommercials!: BesoinCommercial[];
    serviceCommercials!: ServiceCommercial[];
    connexionsEntrantes!: Connexion[];
    entreprise1Reunions?: Reunion[]; // Add this line
    entreprise2Reunions?: Reunion[];
    besoinOffres?: BesoinOffre[];
  }
